src/ceph-disk/ceph_disk/main.py

```diff
@@ -3756,6 +3756,25 @@ def main_activate(args):
             )
             osd_data = get_mount_point(cluster, osd_id)
 
+            args.cluster = cluster
+
+            # Check if encrypted block device in journal
+
+            block_dmcrypt_path = os.path.join(osd_data, 'block_dmcrypt')
+
+            if args.dmcrypt and os.path.exists(block_dmcrypt_path):
+                partition = DevicePartition.factory(
+                    path=None,
+                    dev=os.path.join(osd_data, 'block_dmcrypt'),
+                    args=args)
+
+                partition.osd_dm_key = get_dmcrypt_key(
```

Comment:
Not setting osd_dm_key manually will call the setup_crypt part. However that will not work because the uuid of the partition being unlocked is not the one saved in the key value store of the monitor. Hence the unlock will fail with that uuid.